src/App.tsx

```diff
@@ -5,7 +5,7 @@ import { Counter } from 'components/counter/Counter';
 import { Result } from 'components/result/Result';
 
 export const App: React.FC = () => {
-  const [currentValue, setCurrentValue] = useState<number | null>(null);
+  const [currentValue, setCurrentValue] = useState<number>(0);
```

Comment:
You don't have to specify type of kept value – [it's already inferred](https://www.typescriptlang.org/docs/handbook/type-inference.html)

```suggestion
  const [currentValue, setCurrentValue] = useState(0);
```